feat: circuit element panel component

devartstar opened the pull request:
Fixes: #6 
- Created PanelHeader Vue file - common for all panels which take in the title as a prop and display it.

- Converted the jQuery UI accordion -> [Vuetify expansion panel (variant: accordion)](https://next.vuetifyjs.com/en/components/expansion-panels/)

- Fixed some styles like to match the styles.

- Removed DOM manupulation as string for displaying icons in `setupElementLists()` function in `setup.js`.

- Fetching the data using lifecycle hook `onBeforeMount` and storing it in proper form to display in panel body using for loop.

- Moved `createElement()` function (which creats elemets to drag on the canvas) from `ux.js` to inside vue-componet

- Move the logic of searching elements inside vue-component

**To do:-** 
- [ ] Refactoring of CSS for circuit elements panel

tachyons commented on src/components/Panels/PanelHeader.vue:
```diff
@@ -0,0 +1,13 @@
+
+<script lang="ts" setup>
+const props = defineProps({
+    headerTitle: { type: String, default: 'Panel Header' },
```
Can we make the texts translatable from the beginning itself, it will be difficult to translate later

devartstar replied:
yes sure, I will try to do it.

devartstar replied:
Worked on it, can you please review it.

https://user-images.githubusercontent.com/61665451/175772655-4800403c-62c9-4b73-a1e5-67e7bc319296.mp4

tachyons replied:
fantastic

abstrekt commented on src/components/Extra.vue:
```diff
@@ -978,8 +956,6 @@
-    name: 'Extra',
-}
+<script lang="ts" setup>
+import ElementsPanelVue from './Panels/ElementsPanel/ElementsPanel.vue'
```
lets not use the 'vue' suffix in component names?

devartstar replied:
removed it.

abstrekt commented on src/components/Panels/PanelHeader.vue:
```diff
@@ -0,0 +1,13 @@
+<template>
```
if this is a sharable component between the Panels, let's create a Panels/shared & move it to there, same flow for any shareable file.

devartstar replied:
moved PanelHeader into a folder named shared.

Will also have to make a separate sharable component for `<v-expansion-panel-text>` 
Since it's repeated twice in `ElementsPanel.vue`
And maybe needed once again in `LayoutPanel.vue`

abstrekt commented on src/components/Panels/ElementsPanel/ElementsPanel.vue:
lets enable `https://eslint.org/docs/latest/rules/no-var`, wdyt?

devartstar replied:
If we add this rule we might have to disable it for the `simulators folder`, else it will add up to the linting errors since lots of places use var.
or insead i will use `let` and `const` from now on.

abstrekt commented on src/components/Panels/ElementsPanel/ElementsPanel.vue:
Please make the element search case-insensitive. Currently, I can I'm not able to search for Hex using text "Hex", only "hex" is working.

devartstar replied:
Fixed it.

abstrekt commented on src/components/Panels/ElementsPanel/ElementsPanel.vue:
```diff
@@ -0,0 +1,186 @@
+<template>
+    <div
```
I can experience some lag when opening the panels when trying from the deploy preview, this doesn't seems to be the case on the production instance, can you check?

devartstar replied:
Seemed like it was a common problem for many.
Solved it using this [reference](https://github.com/vuetifyjs/vuetify/issues/10140#issuecomment-605815875)

abstrekt commented:
On initial load, the minimize panel icon on the header seems broken -
![image](https://user-images.githubusercontent.com/42478217/175773842-aa55d207-ea58-4101-867f-4e0ee142d305.png)

There is some padding missing on the button for the last element
![image](https://user-images.githubusercontent.com/42478217/175773771-188eb6dc-f348-4d3b-b9af-6423cff8ca82.png)

abstrekt commented on src/components/Panels/ElementsPanel/ElementsPanel.vue:
```diff
@@ -0,0 +1,186 @@
+window.elementPanelList = []
+
+import { useI18n } from 'vue-i18n'
+const { t, locale } = useI18n()
```
lets make this function global? so that we don't need to import it everywhere?

abstrekt commented on src/styles/css/main.stylesheet.css:
```diff
@@ -500,6 +501,44 @@ input[type='text']:focus {
 }
 
+/* expansion panel styles - start */
+.v-expansion-panel {
```
we shouldn't overwrite library components styles globally, lets's move this to the components style tag, and wrap it inside a custom class?

devartstar replied:
I will fix it next while refactoring the CSS for this component.

devartstar commented:
> On initial load, the minimize panel icon on the header seems broken - 

fixed it, both the maximize and minimize icons were overlapping initially on load giving that weird icon.

> There is some padding missing on the button for the last element

fixed it, margin-bottom to the last panel was not showing.

devartstar commented:
Will refactor the CSS after Layout Elements Panel is also done. (many classes are the same)
